File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/mapping/RequeryMappingContextIntegrationTest.java
package com.coupang.springframework.data.requery.java.mapping;

import com.coupang.springframework.data.requery.core.RequeryOperations;
import com.coupang.springframework.data.requery.java.configs.RequeryTestConfiguration;
import com.coupang.springframework.data.requery.java.domain.basic.BasicUser;
import com.coupang.springframework.data.requery.java.repository.sample.basic.BasicUserRepository;
import com.coupang.springframework.data.requery.mapping.RequeryMappingContext;
import com.coupang.springframework.data.requery.mapping.RequeryPersistentEntity;
import com.coupang.springframework.data.requery.repository.config.EnableRequeryRepositories;
import org.assertj.core.api.Assertions;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.FilterType;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringRunner;

import javax.inject.Inject;

import static org.assertj.core.api.Assertions.*;

/**
 * RequeryMappingContextIntegrationTest
 *
 * @author debop@coupang.com
 * @since 18. 6. 12
 */
@RunWith(SpringRunner.class)
@ContextConfiguration
public class RequeryMappingContextIntegrationTest {

    @Configuration
    @EnableRequeryRepositories(basePackageClasses = { BasicUserRepository.class },
                               includeFilters = @ComponentScan.Filter(value = { BasicUserRepository.class },
                                                                      type = FilterType.ASSIGNABLE_TYPE))
    static class Config extends RequeryTestConfiguration {

    }

    RequeryMappingContext context;

    @Inject BasicUserRepository users;

    @Inject RequeryOperations operations;

    @Before
    public void setup() {
        context = new RequeryMappingContext();
    }

    @Test
    public void setupMappingContextCorrectly() {

        RequeryPersistentEntity<?> entity = context.getRequiredPersistentEntity(BasicUser.class);
        assertThat(entity).isNotNull();
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/mapping/RequeryMappingContextIntegrationTest.kt
package org.springframework.data.requery.kotlin.mapping

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Ignore
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.FilterType
import org.springframework.data.requery.kotlin.configs.RequeryTestConfiguration
import org.springframework.data.requery.kotlin.core.RequeryOperations
import org.springframework.data.requery.kotlin.domain.sample.User
import org.springframework.data.requery.kotlin.repository.config.EnableRequeryRepositories
import org.springframework.data.requery.kotlin.repository.sample.UserRepository
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit4.SpringRunner

@Ignore("현재 Spring Framework에서 무한루프가 발생한다.")
@RunWith(SpringRunner::class)
@ContextConfiguration
class RequeryMappingContextIntegrationTest {

    @Configuration
    @EnableRequeryRepositories(
        basePackageClasses = [UserRepository::class],
        includeFilters = [ComponentScan.Filter(value = [UserRepository::class], type = FilterType.ASSIGNABLE_TYPE)]
    )
    class TestConfig: RequeryTestConfiguration()

    lateinit var context: RequeryMappingContext

    @Autowired lateinit var users: UserRepository
    @Autowired lateinit var operations: RequeryOperations

    @Before
    fun setup() {
        context = RequeryMappingContext()
    }

    @Test
    fun `setup mapping context correctly`() {

        val entity = context.getRequiredPersistentEntity(User::class.java)
        assertThat(entity).isNotNull
    }
}